src/clients/c++/library/request.h.in

```diff
@@ -528,6 +528,16 @@ class InferContext {
     /// \param batch_size The batch size.
     virtual void SetBatchSize(size_t batch_size) = 0;
 
+    /// \return The correlation id for the inferences.
+    virtual CorrelationID CorrelationId() const = 0;
+
+    /// Set the correlation id to use for subsequent inferences.
+    /// \param correlation_id The value of correlation id to be set.
+    /// CorrelationID of the context will only be modified if 
```

Comment:
Description is confusing. "If non-zero this correlation ID overrides the context's correlation ID for all subsequent inference requests".

Reply:
Fixed